### PR TITLE
Remove all C-style for loops

### PR DESCRIPTION
**Pull request by emaloney**
Fixes another set of deprecations, leaving only the `Bit`-related ones.

### CODE REVIEW

File: Sources/CryptoSwift/Array+Extension.swift [hunk 1]
```diff
@@ -12,9 +12,10 @@ extension Array {
     func chunks(chunksize:Int) -> [Array<Element>] {
         var words = [[Element]]()
         words.reserveCapacity(self.count / chunksize)        
-        for var idx = chunksize; idx <= self.count; idx = idx + chunksize {
+        for var idx in chunksize ... self.count {
```
Comment (krzyzanowskim) [inline Sources/CryptoSwift/Array+Extension.swift]: please  don't make space between "..." in ranges. thanks.

File: Sources/CryptoSwift/Rabbit.swift [hunk 2]
```diff
@@ -176,13 +176,16 @@ final public class Rabbit {
         
         var result = [UInt8](count: bytes.count, repeatedValue: 0)
         var output = nextOutput()
-        for var byteIdx = 0, outputIdx = 0; byteIdx < bytes.count; byteIdx += 1, outputIdx += 1 {
+        var outputIdx = 0
+        for byteIdx in 0 ..< bytes.count {
```
Comment (krzyzanowskim) [inline Sources/CryptoSwift/Rabbit.swift]: it lost its readability. Try this:

```swift
bytesIdx = 0
outputIdx = 0

while (byteIdx < bytes.count) {
    bytesIdx += 1
    outputIdx += 1
}
```

File: Sources/CryptoSwift/Rabbit.swift [hunk 3]
```diff
@@ -64,12 +64,12 @@ final public class Rabbit {
         
         // Key divided into 8 subkeys
         var k = [UInt32](count: 8, repeatedValue: 0)
-        for var j = 0; j < 8; j += 1 {
+        for j in 0 ..< 8 {
```
Comment (krzyzanowskim) [inline Sources/CryptoSwift/Rabbit.swift]: and here space... etc.

Comment (emaloney) [inline Sources/CryptoSwift/Rabbit.swift]: That's the way Xcode's fixit does it — will update.

Comment (krzyzanowskim) [inline Sources/CryptoSwift/Rabbit.swift]: Xcode has it's problems ;) thanks.

Comment (emaloney) [inline Sources/CryptoSwift/Rabbit.swift]: Speaking of Xcode problems, would you mind if I submitted a pull request to add `*.xcscmblueprint` to the `.gitignore`?

Whenever I work on this project, Xcode insists on creating the file `CryptoSwift.xcodeproj/project.xcworkspace/xcshareddata/CryptoSwift.xcscmblueprint` which is not in the repo and which I'm constantly deleting.

**Comment by krzyzanowskim** [comment]
Turn out this changes break AES and tests don't pass now.

**Pull request merged**